lib/src/package_analysis/shape.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:meta/meta.dart';
+
+part 'shape.g.dart';
+
+@sealed
+@JsonSerializable()
+class PackageShape {
+  final List<LibraryShape> libraries;
+  final List<ClassShape> classes;
+
+  PackageShape({required this.libraries, required this.classes});
+
+  factory PackageShape.fromJson(Map<String, dynamic> json) =>
+      _$PackageShapeFromJson(json);
+
+  Map<String, dynamic> toJson() => _$PackageShapeToJson(this);
+}
+
+@sealed
+@JsonSerializable()
+class LibraryShape {
+  /// Unique identifier for the library represented by this [LibraryShape]
+  /// within the given [PackageShape]. This uses the syntax beginning with
+  /// `package:` which would usually be used for importing this library.
+  ///
+  /// Example:
+  /// ```
+  /// package:my_package/my_library.dart
+  /// ```
+  final String uri;
+
+  /// `Set` of [ClassShape.id] elements, where each one corresponds to a class
+  /// exported in this library.
+  final Set<int> exportedClasses;
```

Comment:
In my mind, the `exported` prefix tells us that this property uses integer ids as pointers.